android/app/src/main/java/com/microsoft/codepush/react/SettingsManager.java

```diff
@@ -0,0 +1,131 @@
+package com.microsoft.codepush.react;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.facebook.react.bridge.ReadableMap;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class SettingsManager {
+
+    private Context applicationContext;
+
+    public SettingsManager(Context applicationContext) {
+        this.applicationContext = applicationContext;
+    }
+
+    JSONArray getFailedUpdates() {
+        SharedPreferences settings = applicationContext.getSharedPreferences(CodePushConstants.CODE_PUSH_PREFERENCES, 0);
```

Comment:
Should we initialize this object in the constructor? That way each method didn't need to construct it themselves?